### ISSUE TITLE
TODO: request registerScreenshotEventListener callback with captured image result

### ISSUE THREAD

**Issue by krmao** [gbumps/react-native-screenguard#50]
> callback is ok both on android and ios, but need get the screenshot image url, android is ok, but ios is undefined

```
ScreenGuardModule.registerScreenshotEventListener(async result => {
      console.log('- registerScreenshotEventListener result=', result);
```

```
"react-native-screenguard": "^0.3.6",
"react-native": "0.72.4",
```

> may be need got the recent image from Photo library and copy to app cache library and get the url of file, 
>
> or just do a program screenshot again without any permission

**Comment by gbumps** [comment]
remove `async` and you're good to go!

**Comment by krmao** [comment]
> remove `async` and you're good to go!

no, still got 'undefined' @gbumps 

<img width="1003" alt="image" src="https://github.com/gbumps/react-native-screenguard/assets/7344437/34f91fe1-b6d7-47b5-8e83-be2291983e52">

**Comment by gbumps** [comment]
Thanks for reporting @krmao. Currently, there are no result returned from this yet, just callback only. I'm wondering what type of data you want to receive at this point, so I can consider for the next release.

**Comment by krmao** [comment]
for android, should return 'file://' url, shouldn't be 'content://', because for error **Only owner is able to interact with pending/trashed**

a success used example is use RNFS.state(contentUrl) get a 'file://' url from callback 'content://' url

for ios should return a url just like react-native-image-picker return path

```
[{"creationDate": "1708764314", "cropRect": null, "data": null, "duration": null, "exif": null, "filename": "IMG_5983.HEIC", "height": 1080, "localIdentifier": "3FB0208B-D264-43E5-BF1E-2A93E26F6E25/L0/001", "mime": "image/jpeg", "modificationDate": "1708806810", "path": "/private/var/mobile/Containers/Data/Application/7FDB82C3-0626-4B3C-9116-51339383B045/tmp/react-native-image-crop-picker/74D551C9-9D26-43E9-A2BC-0728C4BE3FA1.jpg", "size": 308365, "sourceURL": "file:///var/mobile/Media/DCIM/105APPLE/IMG_5983.HEIC", "uploadingStatus": "waitUpload", "width": 810}]
```

now I maybe use https://github.com/bigzx/react-native-screen-capture/blob/master/ios/ScreenCapture/ScreenCapture.m first, or you got a just copy first and got beta version @gbumps

**Comment by gbumps** [comment]
Well, I can provide you a workaround for this 1st, while waiting for new version: When callback is registered, ask for permission of Gallery, if allowed => get the path of the last file and access its info through 3rd library, if not => show any alert you want.

**Comment by krmao** [comment]
seems it haven’t a valid react native way https://github.com/react-native-image-picker/react-native-image-picker/issues/1377

**Comment by gbumps** [comment]
after thinking about this for a while, I plan to release this feature on the upcoming version. Follow this thread for updates @krmao.

**Comment by krmao** [comment]
thanks for the reply. I will keep watching.

**Comment by gbumps** [comment]
branch `feature/event_with_captured_img_result`

**Comment by krmao** [comment]
very nice ! thank you !

**Comment by gbumps** [comment]
please dont' close, still WIP.

**Comment by gbumps** [comment]
hello @krmao, released in `v0.3.8`. please have a look 

Thanks!

**Comment by krmao** [comment]
Great Job ! Thank you!
